als-actions/shared/src/main/scala/org/mulesoft/als/actions/hover/PatchedHover.scala

```diff
@@ -38,13 +57,53 @@ class PatchedHover(provider: VocabularyProvider) {
   private def buildTerm(meta: ValueType, key: String, dialectName: String): ValueType =
     ValueType(s"http://als.patched/#$dialectName/${meta.name}/$key")
 
-  private val dialectNames: Map[String, String] =
-    Map(
-      Raml10TypesDialect.dialect.id -> "raml10",
-      Raml08TypesDialect.dialect.id -> "raml08",
-      OAS20Dialect.dialect.id       -> "oas2",
-      OAS30Dialect.dialect.id       -> "oas3",
-      AsyncApi20Dialect.dialect.id  -> "asyncapi2"
-    )
+  def indexAliasTerms(valueType: ValueType,
+                      aliases: Aliases,
+                      dialectName: String,
+                      metaName: String): VocabularyProvider = {
+      provider match {
+        case v: AlsVocabularyRegistry => {
+          val hoverProvider = v.branch
+          val generatedTerms: Seq[ClassTerm] = aliases
+            .aliases
+              .map(alias => {
+                val aliasTerm = buildAliasTerm(alias._1, dialectName, metaName)
+                ClassTerm()
+                  .withName(aliasTerm)
+                  .withDisplayName(aliasTerm)
+                  .withDescription(Raml10UsesKeyTerm.description)
+              })
+              .toSeq
+          hoverProvider.index(valueType.ns.base, generatedTerms)
+          hoverProvider
+        }
+        case _ => provider
+      }
+  }
+
+  def buildAliasTerm(name: String, dialectName: String, metaName: String): String = s"$dialectName/${metaName}/$name"
+  
+  private val patches: Seq[() => Unit] = {
+    if (metaUris.isDefined) {
+      Seq(() => bu.annotations.find(classOf[Aliases])
+        .foreach(aliases => indexAliasTerms(valueType, aliases, dialectName, ValueType(metaUris.get).name))
```

Comment:
instead of `metaUris.isDefined` and then `get`, use a map or match

Reply:
done